app/src/sharedTest/java/org/oppia/app/testing/HtmlParserTestActivty.kt

```diff
@@ -0,0 +1,84 @@
+  }
+
+  @Test
+  fun testHtmlContent_handleCustomOppiaTags_parsedHtmldisplaysStyledText() {
+    activityTestRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+    activityTestRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+
+    val textView = activityTestRule.activity.findViewById(R.id.test_html_content_text_view) as TextView
+    val htmlResult: Spannable = HtmlParser(
+      ApplicationProvider.getApplicationContext(), /* entityType= */ "",  /* entityId= */ ""
+    ).parseOppiaHtml(
+      "\u003cp\u003e\"Let's try one last question,\" said Mr. Baker. \"Here's a pineapple cake cut into pieces.\"\u003c/p\u003e\u003coppia-noninteractive-image alt-with-value=\"\u0026amp;quot;Pineapple cake with 7/9 having cherries.\u0026amp;quot;\" caption-with-value=\"\u0026amp;quot;\u0026amp;quot;\" filepath-with-value=\"\u0026amp;quot;pineapple_cake_height_479_width_480.png\u0026amp;quot;\"\u003e\u003c/oppia-noninteractive-image\u003e\u003cp\u003e\u00a0\u003c/p\u003e\u003cp\u003e\u003cstrong\u003eQuestion 6\u003c/strong\u003e: What fraction of the cake has big red cherries in the pineapple slices?\u003c/p\u003e", textView
+    )
+    assertThat(textView.text.toString()).isEqualTo(htmlResult.toString())
+    onView(withId(R.id.test_html_content_text_view)).check(matches(isDisplayed()))
+    onView(withId(R.id.test_html_content_text_view)).check(matches(withText(textView.text.toString())))
+  }
+
+  @Test
+  fun testHtmlContent_nonCustomOppiaTags_notParsed() {
+    activityTestRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+    activityTestRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+
+    val textView = activityTestRule.activity.findViewById(R.id.test_html_content_text_view) as TextView
+    val htmlResult: Spannable = HtmlParser(
+      ApplicationProvider.getApplicationContext(), /* entityType= */ "",  /* entityId= */ ""
+    ).parseOppiaHtml(
+      "\u003cp\u003e\"Let's try one last question,\" said Mr. Baker. \"Here's a pineapple cake cut into pieces.\"\u003c/p\u003e\u003coppia--image alt-with-value=\"\u0026amp;quot;Pineapple cake with 7/9 having cherries.\u0026amp;quot;\" caption-with-value=\"\u0026amp;quot;\u0026amp;quot;\" filepath-value=\"\u0026amp;quot;pineapple_cake_height_479_width_480.png\u0026amp;quot;\"\u003e\u003c/oppia-noninteractive-image\u003e\u003cp\u003e\u00a0\u003c/p\u003e\u003cp\u003e\u003cstrongQuestion 6\u003c/strong\u003e: What fraction of the cake has big red cherries in the pineapple slices?\u003c/p\u003e",
+      textView
+    )
+    // The two strings aren't equal because this HTML contains a Non-Oppia/Non-Html tag e.g. <image> tag and attributes "filepath-value" which isn't parsed.
+    assertThat(textView.text.toString()).isNotEqualTo(htmlResult.toString())
+    onView(withId(R.id.test_html_content_text_view)).check(matches(not(textView.text.toString())))
+  }
+  @After
```

Comment:
Nit: Please add newline above this. Prefer keeping functions separated by 1 empty newline.

Reply:
okay.